main.go

```diff
@@ -167,21 +170,36 @@ func main() {
 //   3. Run data conversion
 //   4. Generate report
 func toSpanner(driver, projectID, instanceID, dbName string, ioHelper *ioStreams, outputFilePrefix string, now time.Time) error {
-	conv, err := schemaConv(driver, ioHelper)
-	if err != nil {
-		return err
-	}
-	// close the seekable file
+	conv := internal.MakeConv()
+	var err error
+	if !dataOnly {
+		conv, err = schemaConv(driver, ioHelper)
+		if err != nil {
+			return err
+		}
+		// close the seekable file
 
-	if ioHelper.seekableIn != nil {
-		defer ioHelper.in.Close()
-	}
+		if ioHelper.seekableIn != nil {
+			defer ioHelper.in.Close()
+		}
 
-	writeSchemaFile(conv, now, outputFilePrefix+schemaFile, ioHelper.out)
-	writeSessionFile(conv, now, outputFilePrefix+sessionFile, ioHelper.out)
-	if schemaOnly {
-		report(driver, nil, ioHelper.bytesRead, "", conv, outputFilePrefix+reportFile, ioHelper.out)
-		return nil
+		writeSchemaFile(conv, now, outputFilePrefix+schemaFile, ioHelper.out)
+		writeSessionFile(conv, now, outputFilePrefix+sessionFile, ioHelper.out)
+		if schemaOnly {
+			report(driver, nil, ioHelper.bytesRead, "", conv, outputFilePrefix+reportFile, ioHelper.out)
+			return nil
+		}
+	} else {
+		// read a session JSON file and unmarshal it's content into
```

Comment:
Suggest factoring out this block (line 193-202) into a separate function, say readSessionFile.
toSpanner is getting quite long now.

Reply:
Done